src/core/segment_buffers/implementations/text/html/text_track_cues_store.ts

```diff
@@ -333,6 +354,18 @@ export default class TextTrackCuesStore {
         }
       }
     }
+
+    if (cuesBuffer.length) {
+      const lastCue = cuesBuffer[cuesBuffer.length - 1];
+      if (areNearlyEqual(lastCue.end, start, relativeDelta)) {
+           // Match the end of the previous cue to the start of
+           // the following one if they are close enough
+           //   ours:                   |AAAAA|
+           //   the current one: |BBBBB|...
+           //   Result:          |BBBBBBBAAAAA|
```

Comment:
Your comments are weirdly sometimes (not the first time) strangely indented more than the code it comments. Is it on purpose?

On another subject, we could explain here that this allows to avoid having a very small gap without subtitles when that was probably not wanted.